server/assistanthandler.go

```diff
@@ -543,67 +547,164 @@ func (h *AssistantHandler) GetUsage(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	err = r.ParseForm()
+	start, end, err := parseDateRangeFromQueryString(r)
 	if err != nil {
+		logger.WithError(err).Error("unable to parse date range from query string")
 		web.Respond(w, r, http.StatusBadRequest, err)
+
 		return
 	}
 
-	zone := r.Form.Get("zone")
-
-	loc, err := time.LoadLocation(zone)
+	usage, err := h.server.Assistantstore.GetUsage(ctx, start, end)
 	if err != nil {
-		log.WithField("timezone", zone).Info("invalid timezone provided by client")
-		loc, _ = time.LoadLocation("UTC")
+		logger.WithError(err).Error("unable to get usage")
+		web.Respond(w, r, http.StatusInternalServerError, err)
+
+		return
 	}
 
-	dateFormat := r.Form.Get("format")
-	rangeParam := r.Form.Get("range")
+	web.Respond(w, r, http.StatusOK, usage)
+}
 
-	rangeParts := strings.Split(rangeParam, " - ")
-	if len(rangeParts) != 2 {
-		logger.WithField("dateRange", rangeParam).Error("invalid date range provided")
-		web.Respond(w, r, http.StatusBadRequest, "invalid date range provided")
+// @Summary      Manage Assistant Sessions
+// @Description  Retrieve a list of all previous chat sessions across all users.
+// @Tags         Assistant
+// @Security     bearer[assistant/read_all]
+// @Produce      json
+// @Success      200  {array}   model.AssistantSession "List of previous conversation sessions"
+// @Failure      401           "Request was not properly authenticated"
+// @Failure      403           "Insufficient permissions for this request"
+// @Failure      500           "Internal SOC error; review SOC logs"
+// @Router       /api/assistant/manage/sessions [get]
+func (h *AssistantHandler) manageSessions(w http.ResponseWriter, r *http.Request) {
+	h.ManageSessions(w, r)
+}
+
+// @Summary      Manage Assistant Sessions
+// @Description  Retrieve a list of all previous chat sessions for a given user.
+// @Tags         Assistant
+// @Security     bearer[assistant/read_all]
+// @Param        userId     path  string              true  "ID of the user to retrieve sessions for"
+// @Produce      json
+// @Success      200  {array}   model.AssistantSession "List of previous conversation sessions"
+// @Failure      401           "Request was not properly authenticated"
+// @Failure      403           "Insufficient permissions for this request"
+// @Failure      500           "Internal SOC error; review SOC logs"
+// @Router       /api/assistant/manage/{userId}/sessions [get]
+func (h *AssistantHandler) manageUserSessions(w http.ResponseWriter, r *http.Request) {
+	h.ManageSessions(w, r)
+}
+
+// @Summary      Manage Assistant Sessions
```

Comment:
Why is this called ManageSessions? Seems like it's more of a RetrieveSessions name.